lib/ddtrace/transport/traces.rb

```diff
@@ -123,7 +123,21 @@ def send_traces(traces)
                 return send_traces(traces)
               end
             end
-          end.force
+          end
+
+          # Force resolution of lazy enumerator.
+          #
+          # The "correct" method to call here would be `#force`,
+          # as this method was created to force the eager loading
+          # of a lazy enumerator.
+          #
+          # Unfortunately, JRuby < 9.2.9.0 erroneously eagerly loads
+          # the lazy Enumerator during intermediate steps.
+          # This forces us to use `#to_a`, as this method works for both
+          # lazy and regular Enumerators.
+          # Using `#to_a` can mask the fact that we expect a lazy
+          # Enumerator.
+          responses = responses.to_a
```

Comment:
Interestingly enough, this was incorrect in TruffleRuby as well, lol: https://github.com/oracle/truffleruby/issues/2273 . Seems like there may be gaps in https://github.com/ruby/spec around this.

Reply:
I think that's accurate, as the JRuby repo got an update to their lazy enum specs, copied from `ruby/spec`, around the same time the bug was fixed.